core/aws_ddk_core/resources/_kinesisfirehose.py

```diff
@@ -0,0 +1,172 @@
+        source_stream: Optional[IStream] = None,
+        **firehose_props: Any,
+    ) -> firehose.IDeliveryStream:
+        """
+        Create and configure Firehose delivery stream.
+
+        This construct allows to configure parameters of the firehose delivery stream using ddk.json
+        configuration file depending on the `environment_id` in which the function is used.
+        Supported parameters are:
+
+        Parameters
+        ----------
+        scope : Construct
+            Scope within which this construct is defined
+        id : str
+            Identifier of the queue
+        environment_id : str
+            Identifier of the environment
+        destinations: Sequence[firehose.IDestination]
+            The destinations that this delivery stream will deliver data to
+        delivery_stream_name: Optional[str] = None
+            A name for the delivery stream
+        encryption: Optional[firehose.StreamEncryption] = None
+            Indicates the type of customer master key (CMK) to use for server-side encryption, if any.
+            Default: StreamEncryption.UNENCRYPTED
+        encryption_key: Optional[IKey] = None
+            Customer managed key to server-side encrypt data in the stream.
+            Default: - no KMS key will be used
+        role: Optional[IRole] = None
+            The IAM role associated with this delivery stream.
+            Assumed by Kinesis Data Firehose to read from sources and encrypt data server-side.
+            Default: - a role will be created with default permissions.
+        source_stream: Optional[IStream] = None
+            The Kinesis data stream to use as a source for this delivery stream
+        **firehose_props: Any
+            Additional properties. For complete list of properties refer to CDK Documentation -
+            Firehose Delivery Stream:
+            https://docs.aws.amazon.com/cdk/api/v1/python/aws_cdk.aws_kinesisfirehose/DeliveryStream.html
+
+        Returns
+        -------
+        delivery_stream: firehose.IDeliveryStream
+            A Kinesis Firehose Delivery Stream
+        """
+        # Collect args
+        firehose_props = {
+            "delivery_stream_name": delivery_stream_name,
+            "destinations": destinations,
+            "encryption": encryption,
+            "encryption_key": encryption_key,
+            "role": role,
+            "source_stream": source_stream,
+            **firehose_props,
+        }
+
+        # create delivery stream
+        _logger.debug(f"firehose_props: {firehose_props}")
+        firehose_stream: firehose.IDeliveryStream = firehose.DeliveryStream(scope, id, **firehose_props)
+
+        return firehose_stream
+
+    @staticmethod
+    def s3_destination(
+        environment_id: str,
+        id: str,
+        bucket: IBucket,
+        buffer_interval: Optional[Duration] = None,
+        buffer_size: Optional[Size] = None,
+        **destination_props: Any,
+    ) -> firehose.IDeliveryStream:
```

Comment:
Should be returning: https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-kinesisfirehose-destinations-alpha.S3Bucket.html not IDeliveryStream